### PR TITLE
Java SDK thread safety and multipart upload download support and test changes

### PR DESCRIPTION
**Pull request by AshwiniBodke**
Done some modifications in AviApi and few test cases

Input json file

Refactored code for naming conventions and best practices.

Added support for file upload and export configuration

Done some modifications in AviApi and few test cases

Refactored code for naming conventions and best practices.

Added few changes for travis.yml

code changes made to make download file functionality generic.

Added support for file upload and export configuration

Done some modifications in AviApi and few test cases

Refactored code for naming conventions and best practices.

Added support for file upload and export configuration

Done some modifications in AviApi and few test cases

Refactored code for naming conventions and best practices.

code changes made to make download file functionality generic.

code refactired for file download

multipart file download support added & additional test cases written for the same

null check added for file downloa

added readme.md and latest jar into the target

added readme.md and latest jar into the target

Did some modifications in test cases

### CODE REVIEW

File: java/test/com/vmware/avi/sdk/MockAviSDKTest.java [hunk 1]
```diff
@@ -2,137 +2,138 @@
...
 			Map<String, String> val = new HashMap<String, String>();
 			val.put("name", name);
 			Mockito.when(aviapi.get("pool", val)).thenReturn(delRes);
 			String msg = delRes.get("error").toString();
 			assertEquals("Pool object not found!", msg);
-	            
-			}
-			catch (AviApiException | ParseException | NullPointerException e) {
+
+		} catch (AviApiException | NullPointerException e) {
```
Comment (chaitanyaavi) [inline java/test/com/vmware/avi/sdk/MockAviSDKTest.java]: Null pointer should not be handled in catch put null check instead.

**Comment by chaitanyaavi** [comment]
Change version in pom.xml to 18.2.8

**Review by chaitanyaavi — CHANGES_REQUESTED** [review]
Please update version and null check changes.

File: java/test/com/vmware/avi/sdk/MockAviSDKTest.java (hunk 1 above)
Comment (AshwiniBodke) [reply java/test/com/vmware/avi/sdk/MockAviSDKTest.java]: Removed.

**Review by chaitanyaavi — APPROVED** [review]

**Review by grastogi23 — APPROVED** [review]

**Pull request merged**